src/tiledquickplugin/mapitem.h

```diff
@@ -41,15 +42,15 @@ class MapItem : public QQuickItem
 {
     Q_OBJECT
 
-    Q_PROPERTY(Tiled::Map *map READ map WRITE setMap NOTIFY mapChanged)
+    Q_PROPERTY(TiledQuick::MapRef map READ map WRITE setMap NOTIFY mapChanged)
```

Comment:
Can you please add a RESET function? That way I can set map to undefined in QML just to be safe, when a level is exited. The function only needs to set mMap.mMap to nullptr.

Reply:
Yes, that's a good alternative to allowing constructing a null `MapRef` instance from QML!

Reply:
I tried assigning null but it gave me an error; probably something to do with it being a gadget, because I know that works with QObject*.

Reply:
Sure, just assigning `null` will not work since it's not a pointer. An alternative to providing an `unsetMap` or `clear` function would be to allow say `mapItem.map = new MapRef()`.